renderer/reducers/pay.js

```diff
@@ -46,8 +52,33 @@ export const SET_REDIRECT_PAY_REQ = 'SET_REDIRECT_PAY_REQ'
  * @param  {{ address }} address Address (payment request)
  * @returns {Function} Thunk
  */
-export const lightningPaymentUri = (event, { address }) => dispatch => {
-  dispatch(setRedirectPayReq({ address }))
+export const lightningPaymentUri = (event, { address }) => (dispatch, getState) => {
+  try {
+    const state = getState()
+
+    // If the user is logged into a wallet, try to autopay the invoice.
+    if (walletSelectors.isWalletOpen(state)) {
+      const autopayList = autopaySelectors.autopayList(state)
+      const invoice = decodePayReq(address)
+      const { payeeNodeKey, satoshis, millisatoshis } = invoice
+      const amountInSats = satoshis || convert('msats', 'sats', millisatoshis)
+      const autopayEntry = autopayList[payeeNodeKey]
+
+      // If there autopay is enabled for the node pubkey we got from the invoice and the amount of the invoice is less
+      // than the autopay's configured limit, pay the invoice silently in the background.
+      if (autopayEntry && amountInSats <= autopayEntry.limit) {
+        dispatch(showAutopayNotification(invoice))
+        return dispatch(payInvoice({ payReq: address, amt: amountInSats }))
+      }
+    }
+
+    // If it wasn't handled with autopay, open in the pay form and focus the app.
+    dispatch(setRedirectPayReq({ address }))
+    dispatch(send('showMainWindow'))
```

Comment:
consider using finally since this is used both in normal and catch routes

Reply:
This is only used in the normal route if not paying with autopay, since the methods aborts after paying with an early return.

Reply:
I reworked this to make it a little easier to follow